docs/compatibility.md

```diff
@@ -48,6 +49,8 @@ request, we are testing that deserialization of the compatibility suite at HEAD
 produces StableHLO programs which are syntactically identical to the original
 StableHLO programs that were serialized by the latest release of libStablehlo.
 
+<a id="future-work"></a>
```

Comment:
I don't think we need explicit anchor tags, given that GitHub autogenerates them anyway? For example, this already works at HEAD: https://github.com/openxla/stablehlo/blob/main/docs/compatibility.md#future-work. (Same below).

Reply:
This is true. There are pros and cons for the explicit anchor tags in a doc like this, they do add clutter. Removed.